qiskit_nature/operators/second_quantization/multiply.py

```diff
@@ -0,0 +1,47 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Mixin for operator scalar multiplication interface.
+"""
+# TODO: Remove this after Terra#5617 is merged.
+
+from abc import ABC, abstractmethod
+
+
+class MultiplyMixin(ABC):
+    """Abstract Mixin for scalar multiplication.
+    This class defines the following operator overloads:
+        - ``*`` / ``__rmul__`
+        - ``/`` / ``__truediv__``
+        - ``__neg__``
+    The following abstract methods must be implemented by subclasses
+    using this mixin
+        - ``_multiply(self, other)``
+    """
+    def __rmul__(self, other):
+        return self._multiply(other)
+
+    def __truediv__(self, other):
+        return self._multiply(1 / other)
+
+    def __neg__(self):
+        return self._multiply(-1)
+
+    @abstractmethod
+    def _multiply(self, other):
+        """Return the {cls} other * self.
+        Args:
+            other (complex): a complex number.
+        Returns:
+            {cls}: the {cls} other * self.
+        """.format(cls=type(self).__name__)
```

Comment:
Okay just one final question: in the `StarAlgebraMixin` you now went with `add` as a public method. Is there a reason for sticking to a private one here? Otherwise we could probably replace `_multiply` with `mul`, right?

Reply:
Yes. But MultiplyMixin is quantum_info's class in the future. I wrote a related comment in the Terra PR. Since Terra's PR is not merged now, the specification here may be changed. If the Terra's PR is merged I will change here.

(Ane the name `mul` is compatible with opflow.)

Reply:
Alright sounds good to me. Once the Terra PR is merged we can directly use the code there rather than adapting the interface here though :+1: